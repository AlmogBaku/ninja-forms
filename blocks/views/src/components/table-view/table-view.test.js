import { Pagination } from "./index";
import { render } from "@testing-library/react";

describe("Pagination", () => {
	const props = {
		previousPage: jest.fn(),
		nextPage: jest.fn(),
		loadMore: jest.fn(),
		more: 2,
		loading: false
	};

	it("Shows button for previous page when it should ", () => {
		const { container } = render(
			<Pagination {...props} canPreviousPage={true} canNextPage={false} />
		);
		expect(container.querySelectorAll("button").length).toBe(1);
		expect(container).toMatchSnapshot();
	});

	it("Shows button for next page when it should ", () => {
		const { container } = render(
			<Pagination {...props} canPreviousPage={false} canNextPage={true} />
		);
		expect(container.querySelectorAll("button").length).toBe(1);
		expect(container).toMatchSnapshot();
	});

	it("Shows neither button when it should show neither button ", () => {
		const { container } = render(
			<Pagination {...props} canPreviousPage={false} canNextPage={false} />
		);
		expect(container.querySelectorAll("button").length).toBe(0);
		expect(container).toMatchSnapshot();
	});
});
