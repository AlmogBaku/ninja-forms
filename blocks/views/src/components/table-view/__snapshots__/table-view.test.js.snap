// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Shows button for next page when it should  1`] = `
<div>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <div />
    <div>
      <button>
        &gt;
      </button>
    </div>
  </div>
</div>
`;

exports[`Pagination Shows button for previous page when it should  1`] = `
<div>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <div>
      <button>
        &lt;
      </button>
    </div>
  </div>
</div>
`;

exports[`Pagination Shows neither button when it should show neither button  1`] = `
<div>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <div />
  </div>
</div>
`;
